### ISSUE TITLE
Suggest String::clone and forbid String::to_string

### ISSUE THREAD

**Issue by MarkusTieger** [rust-lang/rust-clippy#14173]
### What it does

We should add a lint that causes a warning if the to_string method is used on a String / &String, (not on a &str ofc).

### Advantage

The Clone trait is the more commonly used method. Afaik there is no disadvantage of using it instead of ToString.

### Drawbacks

None afaik

### Example

```rust
// 1)
let variable1 = String::new();
let variable2 = variable1.to_string();

// 2)
let variable1 = Some(&variable1);
let variable2 = variable1.map(String::to_string);
```

Could be written as:

```rust
// 1)
let variable1 = String::new();
let variable2 = variable1.clone();

// 2)
let variable1 = Some(String::new());
let variable2 = variable1.cloned();
```

**Comment by Centri3** [comment]
This already exists in the form of `string_to_string`.

**Comment by MarkusTieger** [comment]
> This already exists in the form of `string_to_string`.

Hmm, I see. Would be interested in knowing why this lint isn't enabled by default.
(btw "2)", doesn't get picked up by string_to_string in my example)

**Comment by smoelius** [comment]
Maybe `string_to_string` should be merged into [implicit_clone](https://rust-lang.github.io/rust-clippy/stable/index.html#implicit_clone)?

More people would benefit from it if it were pedantic.

**Comment by MarkusTieger** [comment]
> Maybe `string_to_string` should be merged into [implicit_clone](https://rust-lang.github.io/rust-clippy/stable/index.html#implicit_clone)?
> 
> More people would benefit from it if it were pedantic.

Makes sense
